### PR TITLE
Feature: Pet Item Display

### PR DESCRIPTION
**Pull request by Thunderblade73**
# What
Adds icons for pets for there equipped pet item.
Implemented for Exp Share and Tier Boost implemented. More items can be easily added later on.
## Preview
image image

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/misc/PetExpShareDisplay.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class PetExpShareDisplay {
+
+    @SubscribeEvent
+    fun onRenderItemOverlayPost(event: GuiRenderItemEvent.RenderOverlayEvent.GuiRenderItemPost) {
+        val stack = event.stack ?: return
+        if (!LorenzUtils.inSkyBlock || stack.stackSize != 1) return
+        if (!SkyHanniMod.feature.misc.pets.expShare) return
+
+        val petCandyUsed = stack.getPetItem() ?: return
```
Comment (lunaynx) [inline src/main/java/at/hannibal2/skyhanni/features/misc/PetExpShareDisplay.kt]:
```suggestion
        val petItem = stack.getPetItem() ?: return
```

File: src/main/java/at/hannibal2/skyhanni/features/misc/PetExpShareDisplay.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        if (petCandyUsed != "PET_ITEM_EXP_SHARE") return
```
Comment (lunaynx) [inline src/main/java/at/hannibal2/skyhanni/features/misc/PetExpShareDisplay.kt]:
```suggestion
        if (petItem != "PET_ITEM_EXP_SHARE") return
```

**Review by lunaynx — CHANGES_REQUESTED** [review]

**Review by CalMWolfs — APPROVED** [review]
Config option description is just missing a fullstop but the rest is looking good to me

**Review by CalMWolfs — APPROVED** [review]
worked

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**